pages/docs/couple-your-code/couple-your-code-waveform.md

```diff
@@ -0,0 +1,167 @@
+
+## Linear interpolation in a time window
+
+A simple extension of the existing API is to apply linear interpolation inside of a time window to get smoother coupling boundary conditions. With this approach time-dependent functions (so-called *waveforms*) are exchanged between the participant. Since these waveforms are exchanged iteratively in implicit coupling, we call this procedure *waveform iteration*. Exchanging waveforms leads to a more robust subcycling and allows us to support higher order time stepping (for details, see [here](https://onlinelibrary.wiley.com/doi/full/10.1002/nme.6443)).
+
+### Example for waveform iteration with linear interpolation
+
+Linear interpolation between coupling boundary conditions of the previous and the current time window is illustrated below:
+
+![Coupling data exchange with linear interpolation](images/docs/couple-your-code/couple-your-code-waveform/WaveformLinear.png)
+
+If the Dirichlet participant $$\mathcal{D}$$ calls `readBlockVectorData`, it samples the data from a time-dependent function $$c_\mathcal{D}(t)$$. This function is created from linear interpolation of the first and the last sample $$c_{\mathcal{D}0}$$ and $$c_{\mathcal{D}5}$$ created by the Neumann participant $$\mathcal{N}$$ in the current time window. This allows $$\mathcal{D}$$ to sample the coupling condition at arbitrary times $$t$$ inside the current time window.
+
+## Experimental API for waveform iteration
+
+preCICE offers an experimental API for using linear interpolation on time windows. Here, `readBlockVectorData` accepts an additional argument `dt`. This allows us to choose the time where the interpolant should be sampled:
+
+```cpp
+// stable API with constant data in time window
+void readBlockVectorData(int dataID, int size, const int* valueIndices, double* values) const;
+
+// experimental API for waveform iteration
+void readBlockVectorData(int dataID, int size, const int* valueIndices, double dt, double* values) const;
+```
+
+{% include note.html content="The functionality of `writeBlockVectorData` remains unchanged. All samples but the one from the very last time step in the time window are ignored. An extension of `writeBlockVectorData` to accept multiple samples per time window is work in progress." %}
+
+The experimental API has to be activated in the configuration file via the `experimental` attribute. This allows us to define the order of the interpolant in the `read-data` tag of the corresponding `participant`. Here, `waveform-order="1"` corresponds to linear interpolation and `waveform-order="0"` to constant interpolation (this is just the same as the stable API).
```

Comment:
Sounds complicated. Just say that the default value of `waveform-order` is 0.

Reply:
Changed this in 00373c511500731c2c6af2b262765b5807c51f49. I still think we should mention that `waveform-order="1"` is linear.